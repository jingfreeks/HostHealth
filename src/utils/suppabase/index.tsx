export {default as supabase} from './suppabase';
