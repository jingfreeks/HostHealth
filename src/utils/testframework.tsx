import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
