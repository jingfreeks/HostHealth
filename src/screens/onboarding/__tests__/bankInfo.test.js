import React from 'react';
import {renderWithProviders} from '@/utils/testframeworknew';
import {waitFor} from '@testing-library/react-native';
import {OnBoardingBankInfo} from '../index';

describe('On boarding bankInfo Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<OnBoardingBankInfo />);
    waitFor(() => {
      expect(all.toJSON()).toMatchSnapshot();
    });
  });

});
