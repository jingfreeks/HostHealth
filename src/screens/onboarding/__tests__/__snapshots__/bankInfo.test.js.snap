// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On boarding bankInfo Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "marginBottom": 20,
            "marginTop": 10,
          },
        ]
      }
    />
  </View>
  <View
    style={
      [
        {
          "marginBottom": 30,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        border={10}
        borderc="#99E1E1"
        borderw={2}
        style={
          [
            {
              "borderColor": "#99E1E1",
              "borderRadius": 10,
              "borderWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          accessibilityLabel="BankInfoPreviousButtonTestId"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#d6f3f3",
              "borderRadius": 10,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
          testID="BankInfoPreviousButtonTestId"
        >
          <Text
            style={
              [
                {
                  "color": "#03606A",
                  "fontFamily": "Avenir",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Previous
          </Text>
        </View>
      </View>
      <View
        style={
          [
            {
              "marginRight": 5,
            },
          ]
        }
      />
      <View
        border={10}
        borderc="#99E1E1"
        borderw={2}
        style={
          [
            {
              "borderColor": "#99E1E1",
              "borderRadius": 10,
              "borderWidth": 2,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          accessibilityLabel="BankInfoNextButtonTestId"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#d6f3f3",
              "borderRadius": 10,
              "opacity": 1,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
          testID="BankInfoNextButtonTestId"
        >
          <Text
            style={
              [
                {
                  "color": "#03606A",
                  "fontFamily": "Avenir",
                  "fontSize": 15,
                  "fontWeight": "500",
                },
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
